qiskit/transpiler/passes/routing/sabre_swap.py

```diff
@@ -329,7 +333,9 @@ def _obtain_swaps(self, front_layer, current_layout):
     def _compute_cost(self, layer, layout):
         cost = 0
         for node in layer:
-            cost += self.coupling_map.distance(layout[node.qargs[0]], layout[node.qargs[1]])
+            cost += self.coupling_map.dist_matrix[layout.v2p[node.qargs[0]]][
+                layout.v2p[node.qargs[1]]
+            ]
```

Comment:
Similar comment about tuple v numpy array view as temporary objects.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7036/commits/9268df5513b4965ecf2ea457c8bf86ba697586f1